File: ProyectoCivitasRuby/lib/modelo/sorpresa_pagar_cobrar.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
  class SorpresaPagarCobrar < Sorpresa
    public_class_method :new 
    def initialize(valor, texto)
      super(texto)
      @valor = valor    
    end

    #@Override
    def aplicarajugador(actual, todos)
      if(jugador_correcto(actual, todos))
        informe(actual, todos)
        todos[actual].modificar_saldo(@valor)
      end
    end
    
    def to_s
      salida = "{@texto}\nSorpresa #{@valor} Pagar/Cobrar"
    end
    
  end
end

File: ProyectoCivitasJava/src/civitas/SorpresaPagarCobrar.java
/**
 * Proyecto Civitas
 * Sergio Hervás Cobo
 * Javier Suárez Lloréns
 * Programación y Diseño Orientado a Objetos
 * 2º B3 - Grado en Ingeniería Informática
 * E.T.S.I. Informática y de Telecomunicación
 * Universidad de Granada
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author sergio
 */
public class SorpresaPagarCobrar extends Sorpresa {

    private int valor;

    SorpresaPagarCobrar(int valor, String texto) {
        super(texto);
        this.valor = valor;
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).modificarSaldo(valor);
        }

    }
    
    @Override
    public String toString(){
        String salida = "Sorpresa " + valor + " Pagar/Cobrar";
        return salida;
    }
}
